lib/Storage/S3StorageClient.cpp

```diff
@@ -0,0 +1,379 @@
+/// \file S3StorageClient.cpp
+/// \brief Implementation of StorageClient operations in S3
+
+//
+// This file is distributed under the MIT License. See LICENSE.md for details.
+//
+
+#include <fstream>
+
+#include "llvm/Support/Process.h"
+#include "llvm/Support/YAMLTraits.h"
+
+#include "revng/Storage/Path.h"
+#include "revng/Storage/S3StorageClient.h"
+#include "revng/Support/Assert.h"
+#include "revng/Support/OnQuit.h"
+#include "revng/Support/PathList.h"
+
+#include "aws/core/Aws.h"
+#include "aws/core/auth/AWSCredentials.h"
+#include "aws/core/utils/logging/FormattedLogSystem.h"
+#include "aws/s3/S3Client.h"
+#include "aws/s3/model/CopyObjectRequest.h"
+#include "aws/s3/model/DeleteObjectRequest.h"
+#include "aws/s3/model/GetObjectRequest.h"
+#include "aws/s3/model/HeadObjectRequest.h"
+#include "aws/s3/model/PutObjectRequest.h"
+
+using FilenameMapType = llvm::StringMap<std::string>;
+
+template<>
+struct llvm::yaml::CustomMappingTraits<FilenameMapType> {
+  static void inputOne(IO &IO, llvm::StringRef Key, FilenameMapType &Value) {
+    IO.mapRequired(Key.str().c_str(), Value[Key]);
+  }
+
+  static void output(IO &IO, FilenameMapType &Value) {
+    for (auto &Pair : Value) {
+      IO.mapRequired(Pair.first().str().c_str(), Pair.second);
+    }
+  }
+};
+
+namespace revng::detail {
+
+namespace {
+using Aws::Utils::Logging::FormattedLogSystem;
+using Aws::Utils::Logging::LogLevel;
+
+static Logger<> AWSLogger("aws");
```

Comment:
In `.cpp` files we usually just go for `Log` if it's the default for the `.cpp` file.

Reply:
Renamed to `Logger`, since `Log` is already taken by an exposed AWS SDK function